### PR TITLE
runtime: Forbid claim of unregistered ids - Igor's review

### PR DESCRIPTION
**Pull request by CodeSandwich**
Review comment fixes for https://github.com/radicle-dev/radicle-registry/pull/550

### CODE REVIEW

File: runtime/src/registry.rs [hunk 1]
```diff
@@ -433,35 +433,13 @@ decl_module! {
...
-    Taken,
-
-    /// The Id has been unregistered and is therefore unclaimable
-    Unclaimable,
-}
-
-fn id_availability(id: &Id) -> IdAvailability {
-    use IdAvailability::*;
-
+fn verify_id_availability(id: &Id) -> Result<(), RegistryError> {
```
Comment (geigerzaehler) [inline runtime/src/registry.rs:436]: Suggestion: `ensure_id_is_available` flows better in my opinion.

File: core/src/error.rs [hunk 2]
```diff
@@ -134,9 +131,9 @@ pub enum RegistryError {
 
     #[cfg_attr(
         feature = "std",
-        error("the id has been unregistered and is no longer claimed")
+        error("the ID has been unregistered and can't be claimed again")
     )]
-    UnclaimableId = 18,
+    IdUnregistered = 18,
```
Comment (geigerzaehler) [inline core/src/error.rs:136]: Suggestion: We could use the term “retired” as in `IdRetired`. This is [common terminology](https://en.wikipedia.org/wiki/Retired_number).

File: runtime/src/registry.rs (hunk 1 above)
Comment (CodeSandwich) [reply runtime/src/registry.rs]: Fixed

File: core/src/error.rs (hunk 2 above)
Comment (CodeSandwich) [reply core/src/error.rs]: Love it, fixed

**Pull request merged**